test/hyperduals.jl

```diff
@@ -0,0 +1,68 @@
+function randdual(x)
+  _x = zeros(HyperDualNumbers.Hyper{Float64}, size(x)...)
+  for i in eachindex(x)
+    _x = HyperDualNumbers.Hyper(x[i], rand(), rand(), rand())
+  end
+  return _x
+end
+
+function reinterpretHD(T, A)
+  tmp = reinterpret(T, A)
+  return tmp[1:4:end, :]
```

Comment:
why the slice?

Reply:
reinterpreting a $N \times M$ matrix containing HyperDuals returns a $4N \times M$ matrix containing val, \eps1, \eps2 and \eps12 part of each of the entries. 

This is essentially checking only the val part of the matrix A.

I will do some updates and name the test properly on what they are checking.

Reply:
Don't we also want to check the epsilons?

Reply:
I thought what you mean in https://github.com/JuliaLinearAlgebra/Octavian.jl/pull/179#issuecomment-1594156765 is that we only check val part. I will fix it now.

Reply:
I meant that is what `isapprox` does, so we need to reinterpret to check them all.

Reply:
I now check the entire thing and it should be ready for review. Sorry for iterating so many times.